### PR TITLE
Add connecting event

### PR DESCRIPTION
**Pull request by rufman**
- Emits an event when the websocket is connecting/reconnecting.

- Also makes some options optional 
- Adds source files to the package (so that webpack can optimize the bundle when included in another project).

### CODE REVIEW

File: rws.js [hunk 1]
```diff
@@ -152,6 +155,3 @@ export default class RWS extends EventEmitter {
     }
   }
 }
-
-// For browserify
```
Comment (voidabhi) [inline rws.js:156]: any specific reason for removing this?

File: rws.js [hunk 2]
```diff
@@ -28,9 +30,9 @@ export default class RWS extends EventEmitter {
 
     // Initialize callbacks
     const handlers = [ 'onconnecting', 'onopen', 'onclose', 'onmessage', 'onerror' ];
-    for (const handler of handlers) {
+    handlers.forEach((handler) => {
```
Comment (voidabhi) [inline rws.js]: is this faster?

Comment (rufman) [reply rws.js]: hmm just tested this, `forEach` is slower than using a for loop. will change

File: rws.js (hunk 1 above)
Comment (rufman) [reply rws.js:156]: I'm not sure if browserify needs it. Checking on that

**Comment by voidabhi** [comment]
@rufman are you going to update this pull?

**Comment by rufman** [comment]
Yeah I'm planning on doing this in the next couple days. Sorry for the delay

**Comment by voidabhi** [comment]
No issues, just confirming!

File: rws.js (hunk 1 above)
Comment (rufman) [reply rws.js:156]: I was able to use browserify to compile the file without this

**Comment by rufman** [comment]
@voidabhi Is this ready to merge?

File: rws.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import EventEmitter from 'events';
 
 export default class RWS extends EventEmitter {
-  constructor(url, protocols, options) {
+  constructor(url, options = {}, protocols = []) {
     super();
+
```
Comment (voidabhi) [inline rws.js]: can you remove extra lines

File: rws.js [hunk 4]
```diff
@@ -15,7 +16,9 @@ export default class RWS extends EventEmitter {
     };
     this.url = url;
     this.protocols = protocols;
-    for (const setting in settings) {
+
```
Comment (voidabhi) [inline rws.js]: same as above

File: rws.js (hunk 1 above)
Comment (voidabhi) [reply rws.js:156]: Sorry I missed your Apr 12 update

**Comment by rufman** [comment]
@voidabhi fixed

**Comment by voidabhi** [comment]
@rufman Thanks man!

**Comment by rufman** [comment]
Cool! Could you publish a new version, so I can update my dependencies?

**Comment by voidabhi** [comment]
Done. Check now!

**Pull request merged**